Directory.Build.props

```diff
@@ -157,6 +163,9 @@
     <IsTestAssetProject Condition="$(RepoRelativeProjectDir.Contains('testassets'))">true</IsTestAssetProject>
     <IsSampleProject Condition="$(RepoRelativeProjectDir.Contains('sample'))">true</IsSampleProject>
     <IsAnalyzersProject Condition="$(MSBuildProjectName.EndsWith('.Analyzers'))">true</IsAnalyzersProject>
+    <IsSpecificationTestProject Condition="$(MSBuildProjectName.EndsWith('.Specification.Tests'))">true</IsSpecificationTestProject>
+    <IsToolProject Condition="$(MSBuildProjectName.StartsWith('dotnet-'))">true</IsToolProject>
```

Comment:
We have at least one project in this repo (`dotnet-getdocument`) that is a tool but isn't a _global_ tool. (That one ships only within another package.) Any way to turn this convention off for that project?

Reply:
This convention is good enough for what it's used - to avoid creating reference assembly.

Reply:
The name of the property is unfortunate then. Suggest setting `$(HasReferenceAssembly)` here in a couple of cases. Or, use a slight variant such as `$(NeedsReferenceAssembly)` or `$(_HasReferenceAssembly)`.

Reply:
I'll just make it conditional on OutputType